diff -r c0ee152400ff mod_muc.lua
--- mod_muc.lua	Sun Mar 09 23:48:32 2014 +0100
+++ mod_muc.lua	Tue Mar 25 21:22:55 2014 +0100
@@ -43,6 +43,7 @@
 
 -- Configurable options
 muclib.set_max_history_length(module:get_option_number("max_history_messages"));
+muclib.set_default_history_length(module:get_option_number("default_history_messages"));
 
 module:depends("disco");
 module:add_identity("conference", "text", muc_name);
diff -r c0ee152400ff muc.lib.lua
--- muc.lib.lua	Sun Mar 09 23:48:32 2014 +0100
+++ muc.lib.lua	Tue Mar 25 21:22:55 2014 +0100
@@ -24,7 +24,7 @@
 local md5 = require "util.hashes".md5;
 
 local muc_domain = nil; --module:get_host();
-local default_history_length, max_history_length = 20, math.huge;
+local default_history_length, max_history_length = math.huge, math.huge;
 
 ------------
 local presence_filters = {["http://jabber.org/protocol/muc"]=true;["http://jabber.org/protocol/muc#user"]=true};
@@ -155,7 +155,7 @@
 		if maxchars then maxchars = math.floor(maxchars); end
 
 		local maxstanzas = math.floor(history_tag and tonumber(history_tag.attr.maxstanzas) or #history);
-		if not history_tag then maxstanzas = 20; end
+		if not history_tag then maxstanzas = default_history_length; end
 
 		local seconds = history_tag and tonumber(history_tag.attr.seconds);
 		if seconds then seconds = datetime.datetime(os.time() - math.floor(seconds)); end
@@ -1139,6 +1139,13 @@
 	max_history_length = _max_history_length or math.huge;
 end
 
+function _M.set_default_history_length(_default_history_length)
+	default_history_length = _default_history_length or 20;
+	if default_history_length > max_history_length then
+		default_history_length = max_history_length;
+	end
+end
+
 _M.room_mt = room_mt;
 
 return _M;
